### ISSUE TITLE
CLI tool throws NullPointerException from time to time

### ISSUE THREAD

**Issue by kshepitko** [allure-framework/allure-cli#33]
I'm using Allure CLI tool 2.3 to generate HTML reports. This is what I get from time to time:
```
[root@myhost ~]# sudo /opt/allure-cli/bin/allure --debug generate -v 1.4.9 /var/log/te_logs/b28434fb-dae0-49ec-926c-663e2f5d6ba7 -o ./gened
Generating report for Allure version [1.4.9]
Processing result pattern [/var/log/te_logs/b28434fb-dae0-49ec-926c-663e2f5d6ba7]
Found results directory [/var/log/te_logs/b28434fb-dae0-49ec-926c-663e2f5d6ba7]
log4j:ERROR A "org.apache.log4j.ConsoleAppender" object is not assignable to a "org.apache.log4j.Appender" variable.
log4j:ERROR The class "org.apache.log4j.Appender" was loaded by
log4j:ERROR [java.net.URLClassLoader@6a2fc45d] whereas object of type
log4j:ERROR "org.apache.log4j.ConsoleAppender" was loaded by [sun.misc.Launcher$AppClassLoader@43059849].
log4j:ERROR Could not instantiate appender named "console".
Successfully generated report to [/root/./gened].
```
In 2 minutes:
```
[root@myhost  ~]# sudo /opt/allure-cli/bin/allure --debug generate /var/log/te_logs/b28434fb-dae0-49ec-926c-663e2f5d6ba7 -o ./gened
Generating report for Allure version [1.3.9]
Processing result pattern [/var/log/te_logs/b28434fb-dae0-49ec-926c-663e2f5d6ba7]
Found results directory [/var/log/te_logs/b28434fb-dae0-49ec-926c-663e2f5d6ba7]
Connection request: [route: {}->http://repo1.maven.org][total kept alive: 0; route allocated: 0 of 50; total allocated: 0 of 100]
Connection leased: [id: 0][route: {}->http://repo1.maven.org][total kept alive: 0; route allocated: 1 of 50; total allocated: 1 of 100]
Connecting to repo1.maven.org:80
Connection org.apache.http.impl.conn.DefaultClientConnection@77a10bd7 closed
Connection org.apache.http.impl.conn.DefaultClientConnection@77a10bd7 shut down
Connection org.apache.http.impl.conn.DefaultClientConnection@77a10bd7 closed
Connection released: [id: 0][route: {}->http://repo1.maven.org][total kept alive: 0; route allocated: 0 of 50; total allocated: 0 of 100]
java.lang.NullPointerException
ru.yandex.qatools.allure.report.AllureReportBuilderException: java.lang.NullPointerException
```
The problem is that the server is quite flaky, but NPE is quite a strange reaction to the environment issues. Perhaps the problem is caused by a networking issue, but it's not being reported by CLI tool so, therefore impossible to find out the reason (even stack trace in the log is missing) and address it.

**Comment by vania-pooh** [comment]
@kshepitko I would agree that this seems to be a connection-related issue.

**Comment by volkovs** [comment]
Might be similar NPE at Aether (generating via Jenkins plugin):
`
Allure Report Generation: started
Allure Report Generation: find directories by mask [**/allure-results]
Allure Report Generation: found allure result directories [/home/ossrcdm/jenkins/workspace/VAPP_PCJ2/target/allure-results]
Allure Report Generation: copy founded directories in directory [/home/ossrcdm/jenkins/workspace/VAPP_PCJ2/allure3911849477814031594.tmp/results]
Allure Report Generation: generate report from directory [/home/ossrcdm/jenkins/workspace/VAPP_PCJ2/allure3911849477814031594.tmp]
Allure Report Generation: proxy settings [active:'false', host:'null', port:'0', username:'null', password: '']
ERROR: Publisher ru.yandex.qatools.allure.jenkins.AllureReportPublisher aborted due to exception
java.io.IOException: remote file operation failed: /home/ossrcdm/jenkins/workspace/VAPP_PCJ2/allure3911849477814031594.tmp at hudson.remoting.Channel@7a137067:Jenkins_ORCH_UpgradeFlow_61: java.io.IOException: ru.yandex.qatools.allure.report.AllureReportBuilderException: java.lang.NullPointerException
	at hudson.FilePath.act(FilePath.java:985)
	at hudson.FilePath.act(FilePath.java:967)
	at ru.yandex.qatools.allure.jenkins.AllureReportPublisher.generateReport(AllureReportPublisher.java:267)
	at ru.yandex.qatools.allure.jenkins.AllureReportPublisher.perform(AllureReportPublisher.java:146)
	at hudson.tasks.BuildStepMonitor$1.perform(BuildStepMonitor.java:20)
	at hudson.model.AbstractBuild$AbstractBuildExecution.perform(AbstractBuild.java:761)
	at hudson.model.AbstractBuild$AbstractBuildExecution.performAllBuildSteps(AbstractBuild.java:721)
	at hudson.model.Build$BuildExecution.post2(Build.java:183)
	at hudson.model.AbstractBuild$AbstractBuildExecution.post(AbstractBuild.java:670)
	at hudson.model.Run.execute(Run.java:1766)
	at hudson.model.FreeStyleBuild.run(FreeStyleBuild.java:43)
	at hudson.model.ResourceController.execute(ResourceController.java:98)
	at hudson.model.Executor.run(Executor.java:374)
Caused by: java.io.IOException: ru.yandex.qatools.allure.report.AllureReportBuilderException: java.lang.NullPointerException
	at ru.yandex.qatools.allure.jenkins.utils.ReportGenerator.invoke(ReportGenerator.java:50)
	at ru.yandex.qatools.allure.jenkins.utils.ReportGenerator.invoke(ReportGenerator.java:25)
	at hudson.FilePath$FileCallableWrapper.call(FilePath.java:2688)
	at hudson.remoting.UserRequest.perform(UserRequest.java:121)
	at hudson.remoting.UserRequest.perform(UserRequest.java:49)
	at hudson.remoting.Request$2.run(Request.java:325)
	at hudson.remoting.InterceptingExecutorService$1.call(InterceptingExecutorService.java:68)
	at java.util.concurrent.FutureTask$Sync.innerRun(FutureTask.java:334)
	at java.util.concurrent.FutureTask.run(FutureTask.java:166)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
	at java.lang.Thread.run(Thread.java:722)
	at ......remote call to Jenkins_ORCH_UpgradeFlow_61(Native Method)
	at hudson.remoting.Channel.attachCallSiteStackTrace(Channel.java:1360)
	at hudson.remoting.UserResponse.retrieve(UserRequest.java:221)
	at hudson.remoting.Channel.call(Channel.java:753)
	at hudson.FilePath.act(FilePath.java:978)
	... 12 more
Caused by: ru.yandex.qatools.allure.report.AllureReportBuilderException: java.lang.NullPointerException
	at ru.yandex.qatools.allure.report.AllureReportBuilder.processResults(AllureReportBuilder.java:133)
	at ru.yandex.qatools.allure.jenkins.utils.ReportGenerator.invoke(ReportGenerator.java:47)
	at ru.yandex.qatools.allure.jenkins.utils.ReportGenerator.invoke(ReportGenerator.java:25)
	at hudson.FilePath$FileCallableWrapper.call(FilePath.java:2688)
	at hudson.remoting.UserRequest.perform(UserRequest.java:121)
	at hudson.remoting.UserRequest.perform(UserRequest.java:49)
	at hudson.remoting.Request$2.run(Request.java:325)
	at hudson.remoting.InterceptingExecutorService$1.call(InterceptingExecutorService.java:68)
	at java.util.concurrent.FutureTask$Sync.innerRun(FutureTask.java:334)
	at java.util.concurrent.FutureTask.run(FutureTask.java:166)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
	at java.lang.Thread.run(Thread.java:722)
Caused by: java.lang.NullPointerException
	at org.eclipse.aether.internal.impl.DefaultRepositorySystem.resolveDependencies(DefaultRepositorySystem.java:361)
	at ru.yandex.qatools.clay.Aether.resolveWithTransitives(Aether.java:177)
	at ru.yandex.qatools.clay.Aether.resolve(Aether.java:131)
	at ru.yandex.qatools.clay.Aether.resolve(Aether.java:109)
	at ru.yandex.qatools.allure.report.AllureReportBuilder.processResults(AllureReportBuilder.java:127)
	... 12 more
`

**Comment by vania-pooh** [comment]
@volkovs anyway this seems to be an issue in third-party library.

**Comment by kshepitko** [comment]
@vania-pooh any progress on this issue?

**Comment by vania-pooh** [comment]
@kshepitko not yet as we consider this a minor issue. But pull requests are welcome.

**Comment by ilyin371** [comment]
NPE is indeed caused by Aether at `org.eclipse.aether.internal.impl.DefaultRepositorySystem.resolveDependencies(DefaultRepositorySystem.java:361)`. It occurs when remote repository is not reachable (due to network issues). Added pull request to include the stack trace in CLI tool output as well.

**Comment by vania-pooh** [comment]
Fixed in #34.

**Comment by kshepitko** [comment]
The minor problem with stack trace is resolved, but the major problem with NPE is not.
So now it's clear that in both cases - CLI tool and Jenkins plugin - Eclipse Aether is in charge.
Any plans regarding this? Upgrade/downgrade to another Aether version, probably?
